test/models/merchant_test.rb

```diff
@@ -0,0 +1,74 @@
+require "test_helper"
+
+describe Merchant do
+  # let(:merchant) { Merchant.new }
+
+  describe 'validations on merchant' do
+    it 'requires a username to be valid' do
+      #merchant with no username is bad
+      merchant = Merchant.new(email: 'hey@ada.org', uid: 19, provider: "hi")
+      merchant.valid?.must_equal false
+      merchant.errors.messages.must_include :username
+      #merchant with username is good
+      merchants(:dan).valid?.must_equal true
```

Comment:
These should really be two separate tests. One is testing the validation and the other is testing that it would work